docs/handbooks/language-implementation.md

```diff
@@ -0,0 +1,144 @@
+# Language Implementation Handbook
+
+## Foreword
+
+Implementing a new language in *jsii* is not just a matter of implementing code
+generation: mapping the *[jsii type system]* to a new programming language takes
+some amounts of API design, finding ways to represent APIs in the most idiomatic
```

Comment:
Either prephase what API is or use a different/more appropriate term here. Maybe 'language interface design' ?